manager/assets/modext/widgets/core/modx.grid.js

```diff
@@ -495,14 +515,190 @@ Ext.extend(MODx.grid.Grid,Ext.grid.EditorGridPanel,{
+     *
+     * @return void
+     */
+    ,setEditableColumnAccess: function(columnIds) {
+        if (!this.userCanEdit && !Ext.isEmpty(columnIds)) {
+            const colModel = this.getColumnModel();
+            columnIds = columnIds.map(item => item.trim());
+            columnIds.forEach(colId => {
+                const colIndex = colModel.getIndexById(colId);
+                colModel.setEditable(colIndex, false);
+            });
+        }
+    }
+
+
+    /* User Group-Level Permissions Checks for the calling "class" object */
+
+    /**
+     * @property {Function} setUserCanEdit - Assigns a value to userCanEdit property based on
+     * the user's permissions; used to adjust which menu items are available, whether to render links
+     * to and item's editing page, and css cues across many grid classes
+     *
+     * @param {Array} permissions - A set of permissions keys to evaluate; note that many areas currently
+     * rely on a pair of permissions (save_x and edit_x), both of which must be enabled to edit a grid item
+     *
+     * @return void
+     */
+    ,setUserCanEdit: function(permissions) {
+        permissions = permissions.map(item => item.trim());
+        this.userCanEdit = permissions.every(permission => MODx.perm[permission]);
+    }
+
+    /**
+     * @property {Function} setUserCanCreate - Assigns a value to userCanCreate property based on
+     * the user's permissions; used to adjust which menu items are available (namely the Duplicate item)
+     * and whether to render the Create button in the grid's toolbar
+     *
+     * @param {Array} permissions - A set of permissions keys to evaluate; note that many areas currently
+     * rely on a pair of permissions (save_x and new_x), both of which must be enabled to create/duplicate a grid item
+     *
+     * @return void
+     */
+    ,setUserCanCreate: function(permissions) {
+        permissions = permissions.map(item => item.trim());
+        this.userCanCreate = permissions.every(permission => MODx.perm[permission]);
+    }
+
+    /**
+     * @property {Function} setUserCanDelete - Assigns a value to userCanDelete property based on
+     * the user's permissions; used to adjust which menu items are available in the context menus
+     * and whether to render the Delete menu item within a grid toolbar's Batch button
+     *
+     * @param {Array} permissions - A set of permissions keys to evaluate
+     *
+     * @return void
+     */
+    ,setUserCanDelete: function(permissions) {
+        permissions = permissions.map(item => item.trim());
+        this.userCanDelete = permissions.every(permission => MODx.perm[permission]);
+    }
+
+
+    /* Record-Level Permissions Checks, for objects with specific policies */
+
+    ,userCanEditRecord: function(record) {
+        const permissions = record.json.permissions;
+        return !Ext.isEmpty(permissions) && permissions.update === true ? true : false ;
```

Comment:
```suggestion
        return !Ext.isEmpty(permissions) && permissions.update;
```

Reply:
I know the truthy check works, but I made a conscious decision to check for strict true. Leaving the way this is done in the three methods for now...